grub-core/loader/multiboot_mbi2.c

```diff
@@ -708,21 +858,34 @@ grub_multiboot2_make_mbi (grub_uint32_t *target)
   grub_err_t err;
   grub_size_t bufsize;
   grub_relocator_chunk_t ch;
+  struct grub_slaunch_params *slparams = grub_slaunch_params();
 
   bufsize = grub_multiboot2_get_mbi_size ();
 
   COMPILE_TIME_ASSERT (MULTIBOOT_TAG_ALIGN % sizeof (grub_properly_aligned_t) == 0);
 
-  err = grub_relocator_alloc_chunk_align (grub_multiboot2_relocator, &ch,
-					  MBI_MIN_ADDR, 0xffffffff - bufsize,
-					  bufsize, MULTIBOOT_TAG_ALIGN,
-					  GRUB_RELOCATOR_PREFERENCE_NONE, 1);
+  if (grub_slaunch_platform_type () == SLP_INTEL_TXT)
+    {
+      /* MBI was saved and allocated earlier for slaunch */
+      *target = slparams->boot_params_addr;
+      /* For x86 and x64 the allocated physical addr of chunk is the same as virtual */
```

Comment:
Proof? IIRC it's only the same if it can be allocated at target address, otherwise it can be different and moved by relocator just before jump to kernel.

Reply:
https://github.com/TrenchBoot/grub/blob/master/grub-core/lib/relocator.c#L92
https://github.com/TrenchBoot/grub/blob/master/grub-core/lib/relocator.c#L98
https://github.com/TrenchBoot/grub/blob/master/grub-core/lib/relocator.c#L1467 and thereabouts.

Reply:
Remvoed. Now saving the target MBI physical address into boot_params_addr. THe MBI allocation is done as previously